### PR TITLE
Add test(s) for the optional_redirect example

### PR DESCRIPTION
**Pull request by sethlopezme**
Relates to #62.

### CODE REVIEW

File: examples/optional_redirect/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#[test]
+fn test_200() {
+    test("/users/Sergio", Status::Ok, Some("Hello, Sergio!".to_string()));
+    test("/users/login",
+         Status::Ok,
+         Some("Hi! That user doesn't exist. Maybe you need to log in?".to_string()));
+}
+
+#[test]
+fn test_303() {
```
Comment (SergioBenitez) [inline examples/optional_redirect/src/tests.rs]: These should check the `Location` header to ensure the redirect goes to the right place. Something like:

```rust
let location: Vec<_> = response.header_values("Location").collect();
assert_eq!(location, vec!["/users/login"]);
```

Comment (sethlopezme) [inline examples/optional_redirect/src/tests.rs]: Good catch, I didn't think to test the Location header. Tests have been updated!

File: examples/optional_redirect/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+use super::rocket;
+use rocket::testing::MockRequest;
+use rocket::http::{Method, Status};
+
+fn test_200(uri: &str, expected_status: Status, expected_body: String) {
```
Comment (SergioBenitez) [inline examples/optional_redirect/src/tests.rs]: This function is called `test_200`, so you shouldn't need the `expected_status` parameter. Test for Status::Ok directly below instead.

File: examples/optional_redirect/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    let rocket = rocket::ignite()
+        .mount("/", routes![super::root, super::user, super::login]);
+    let mut request = MockRequest::new(Method::Get, uri);
+    let mut response = request.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), expected_status);
+    assert_eq!(response.body().and_then(|b| b.into_string()), Some(expected_body));
+}
+
+fn test_303(uri: &str, expected_status: Status, expected_location: &str) {
```
Comment (SergioBenitez) [inline examples/optional_redirect/src/tests.rs]: This function is called `test_303`, so you shouldn't need the `expected_status` parameter. Test for Status::SeeOther directly below instead.

File: examples/optional_redirect/src/tests.rs (hunk 2 above)
Comment (SergioBenitez) [inline examples/optional_redirect/src/tests.rs]: Take `expected_body` as an `&str` and do a `to_string` inside this function.

Comment (sethlopezme) [inline examples/optional_redirect/src/tests.rs]: Good catch. Didn't mean to leave this in here.

File: examples/optional_redirect/src/tests.rs (hunk 3 above)
Comment (sethlopezme) [inline examples/optional_redirect/src/tests.rs]: Good catch. Didn't mean to leave this in here.

**Comment by SergioBenitez** [comment]
Merged in 9580d6cdfde8cdf947d9be3ea7186cb2680fae00.

**Pull request closed without merging**